src/features/session_trace/aggregate/trace/storage.js

```diff
@@ -164,8 +164,12 @@ export class TraceStorage {
       if (!(typeof val === 'number' && val >= 0)) continue
 
       val = Math.round(val)
-      if (this.parent.timeKeeper && this.parent.timeKeeper.ready && val >= this.parent.timeKeeper.correctedOriginTime) {
-        val = this.parent.timeKeeper.convertAbsoluteTimestamp(val)
+      if (this.parent.timeKeeper && this.parent.timeKeeper.ready && isAbsoluteTimestamp) {
+        val = this.parent.timeKeeper.convertAbsoluteTimestamp(
+          Math.floor(this.parent.timeKeeper.correctAbsoluteTimestamp(
+            this.parent.timeKeeper.correctAbsoluteTimestamp(val)
```

Comment:
is this double invokation of `correctAbsoluteTimestamp` intended here

Reply:
oops